packages/components/button/src/__tests__/Button.spec.tsx

```diff
@@ -1,15 +1,192 @@
+
+    const buttonElement = screen.getByRole('button', {
+      name: /Primary button and disabled/i,
+    });
+    expect(buttonElement).toBeDisabled();
+  });
+  it('should render a XL clickable primary button', () => {
+    renderTheBtnType('Primary button and available', 'primary', false, 'xl');
+
+    const buttonElement = screen.getByRole('button', {
+      name: /Primary button and available/i,
+    });
+    expect(buttonElement).not.toBeDisabled();
+  });
+  it('shoud render the click on primary button', () => {
+    const handleClick = jest.fn();
+    renderTheBtnType(
+      'Primary button and available',
+      'primary',
+      false,
+      'lg',
+      handleClick,
+    );
+
+    const buttonElement = screen.getByRole('button', {
+      name: /Primary button and available/i,
+    });
+    fireEvent.click(buttonElement);
+    expect(handleClick).toHaveBeenCalledTimes(1);
+  });
+  it('should render a clickable secondary button', () => {
+    renderTheBtnType('Secondary button and available', 'secondary', false);
+
+    const buttonElement = screen.getByRole('button', {
+      name: /Secondary button and available/i,
+    });
+    expect(buttonElement).not.toBeDisabled();
+  });
+  it('should render a SM clickable secondary button', () => {
+    renderTheBtnType(
+      'Secondary button and available',
+      'secondary',
+      false,
+      'sm',
+    );
+
+    const buttonElement = screen.getByRole('button', {
+      name: /Secondary button and available/i,
+    });
+    expect(buttonElement).not.toBeDisabled();
+  });
+  it('should render a disabled secondary button', () => {
+    renderTheBtnType('Secondary button and disabled', 'secondary', true);
+
+    const buttonElement = screen.getByRole('button', {
+      name: /Secondary button and disabled/i,
+    });
+    expect(buttonElement).toBeDisabled();
+  });
+  it('shoud render the click on secondary button', () => {
+    const handleClick = jest.fn();
+    renderTheBtnType(
+      'Secondary button and available',
+      'secondary',
+      false,
+      'lg',
+      handleClick,
+    );
+
+    const buttonElement = screen.getByRole('button', {
+      name: /Secondary button and available/i,
+    });
+    fireEvent.click(buttonElement);
+    expect(handleClick).toHaveBeenCalledTimes(1);
+  });
+  it('should render a clickable tertiary button', () => {
+    renderTheBtnType('Tertiary button and available', 'tertiary', false);
+
+    const buttonElement = screen.getByRole('button', {
+      name: /Tertiary button and available/i,
+    });
+    expect(buttonElement).not.toBeDisabled();
+  });
+  it('should render a MD clickable tertiary button', () => {
+    renderTheBtnType('Tertiary button and available', 'tertiary', false, 'md');
+
+    const buttonElement = screen.getByRole('button', {
+      name: /Tertiary button and available/i,
+    });
+    expect(buttonElement).not.toBeDisabled();
+  });
+  it('should render a disabled tertiary button', () => {
```

Comment:
I think you are repeating some tests, there is no real difference between a primary or secondary when testing a click. The behavior is the same depending on the type.